File: API/Models/Pizza.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{
    [Table("Pizzas")]
    public class Pizza
    {
        [Key]
        public long Id { get; set; }

        [MaxLength(100)]
        public required string Name { get; set; }
        public double Weight { get; set; }
        public double Calories { get; set; }
        public double Price { get; set; }

        public bool isDeleted { get; set; } = false;

        public virtual ICollection<PizzaIngredient> PizzaIngredients { get; set; } = new List<PizzaIngredient>();
        public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();
    }
}

File: GUI/src/types/pizza.ts
export interface PizzaIngredient {
  ingredientId: number;
  ingredientName: string;
  ingredientAmount: number;
  price: number;
  weight: number;
  calories: number;
}

export interface Pizza {
  id: number;
  name: string;
  price: number;
  ingredients: PizzaIngredient[];
}

export interface FullPizza {
  id: number;
  name: string;
  price: number;
  weight: number;
  calories: number;
  quantity: number;
  ingredients: PizzaIngredient[];
}

export interface CartItemStored {
  id: number;
  quantity: number;
}
